html/form/basic.php

```diff
@@ -1,16 +1,21 @@
-<form id="smly" class="container" action="#" method="post">
+<form id="smly" class="container" action="<?php echo admin_url('admin-post.php'); ?>" method="post">
+	<?php if ( $this->api_credentials === '' ) { ?>
+		<p class="no_auth" style="padding:15px;background-color:#f2dede;margin:0 0 10px"><?php echo esc_html__( 'Smaily credentials not validated. Subscription form will not work!', 'wp_smaily'); ?></p>
+	<?php } ?>
+	<?php if ( !empty( $_GET['smaily_form_error'] ) ) { ?>
+		<p class="error-nojs" style="padding:15px;background-color:#f2dede;margin:0 0 10px"><?php echo htmlspecialchars( $_GET['smaily_form_error'] ) ?></p>
+	<?php } ?>
 	<p class="error" style="padding:15px;background-color:#f2dede;margin:0 0 10px;display:none"></p>
```

Comment:
This template becomes awfully busy and difficult to follow. I think we should refactor it to something like:
```suggestion
	<p class="error" style="padding:15px;background-color:#f2dede;margin:0 0 10px;display:<?php echo $this->form_has_error ? 'block' : 'none'; ?>"><?php echo esc_html( $this->error_message ); ?></p>
```

And provide necessary template variables from `Smaily_Newsletter_Subscription_Widget::widget` method:
```php
// Create admin template.
// ...

$form_has_error = false;
$error_message = null;
if ( isset( $_GET['smaily_form_error'] ) && !empty( $_GET['smaily_form_error'] ) ) {
    $form_has_error = true;
    $error_message = $_GET['smaily_form_error'];
}
elseif ( !isset( $config['api_credentials'] ) || empty( $config['api_credentials'] ) ) {
    $form_has_error = true;
    $error_message = __( 'Smaily credentials not validated. Subscription form will not work!', 'wp_smaily' );
}

$template->assign( array(
    'form_has_error' => $form_has_error,
    'error_message' => $error_message,
) );

// Render template.
// ...
```

This removes error detection logic from template into backend code, which makes it more manageable in the future.

PS. Same should be applied to `advanced.php` template.